src/xrdsvc/HttpMonitorModule.cc

```diff
@@ -0,0 +1,123 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "xrdsvc/HttpMonitorModule.h"
+
+// System headers
+#include <set>
+#include <stdexcept>
+
+// Qserv headers
+#include "http/Exceptions.h"
+#include "http/RequestQuery.h"
+#include "mysql/MySqlUtils.h"
+#include "wbase/FileChannelShared.h"
+#include "wbase/TaskState.h"
+#include "wconfig/WorkerConfig.h"
+#include "wcontrol/Foreman.h"
+#include "wcontrol/ResourceMonitor.h"
+#include "wpublish/QueriesAndChunks.h"
+
+using namespace std;
+using json = nlohmann::json;
+
+namespace lsst::qserv::xrdsvc {
+
+void HttpMonitorModule::process(string const& context, shared_ptr<wcontrol::Foreman> const& foreman,
+                                shared_ptr<qhttp::Request> const& req,
+                                shared_ptr<qhttp::Response> const& resp, string const& subModuleName,
+                                http::AuthType const authType) {
+    HttpMonitorModule module(context, foreman, req, resp);
+    module.execute(subModuleName, authType);
+}
+
+HttpMonitorModule::HttpMonitorModule(string const& context, shared_ptr<wcontrol::Foreman> const& foreman,
+                                     shared_ptr<qhttp::Request> const& req,
+                                     shared_ptr<qhttp::Response> const& resp)
+        : HttpModule(context, foreman, req, resp) {}
+
+json HttpMonitorModule::executeImpl(string const& subModuleName) {
+    string const func = string(__func__) + "[sub-module='" + subModuleName + "']";
+    debug(func);
+    enforceInstanceId(func, wconfig::WorkerConfig::instance()->replicationInstanceId());
+    enforceWorkerId(func);
+    if (subModuleName == "CONFIG")
+        return _config();
+    else if (subModuleName == "MYSQL")
+        return _mysql();
+    else if (subModuleName == "STATUS")
+        return _status();
+    else if (subModuleName == "ECHO")
+        return _echo();
+    throw invalid_argument(context() + func + " unsupported sub-module");
```

Comment:
I think it's probably worth making and enum with to and from string functions for subModuleName and a switch statement here.

Reply:
Same reply as for your comment in the module `HttpReplicaMgtModule.cc` - I will in a separate effort.